src/memray/_memray/tracking_api.cpp

```diff
@@ -441,16 +480,43 @@ Tracker::childFork()
         return;
     }
 
+    assert(old_tracker != nullptr);
+
     // Re-enable tracking with a brand new tracker.
     d_instance_owner.reset(new Tracker(
             std::move(new_writer),
             old_tracker->d_unwind_native_frames,
             old_tracker->d_memory_interval,
             old_tracker->d_follow_fork,
-            old_tracker->d_trace_python_allocators));
+            old_tracker->d_trace_python_allocators,
+            old_tracker->d_sampling_interval));
     RecursionGuard::isActive = false;
 }
 
+// static std::unordered_set<void*> pointers;
+
+static constexpr char TRACE_MARK = 42;
```

Comment:
I'd suggest that a better mark is a pointer to itself. That is:
```c
*(void*)(addr) = (void*)addr;
```

I think that's _extremely_ unlikely to occur by chance. It does require us to think about alignment, though.

Reply:
Let's stick with just the 1 byte of overhead for now, since it's simpler and means we don't need to worry about alignment. We can do this in a follow up PR if we want to.